tests/Query.test.ts

```diff
@@ -322,6 +322,95 @@ describe('Query', () => {
         );
     });
 
+    it('filters based off a single tag', () => {
```

Comment:
There's a lot of boilerplate and repetition here. As a general comment, there is huge value in tests being easy to read, easy to understand the intention, easy to compare with other related tests to spot differences, and easy to write.

In case you haven't noticed, please have a look at the simplifications I made above, such as in 'filtering with "tag" and global task filter'.

These new tests can be made similarly much simpler.

Reply:
Covered as part of test refactor.